app/controllers/users_controller.rb

```diff
@@ -52,6 +52,20 @@ def redirect
 
   private
 
+  def set_cookie_and_redirect(email)
+    cookies[:h_email] = { value: email }
+    redirect_to '/refer-a-friend'
+  end
+
+  def user_exists
+    email = params[:user][:email].downcase
+    params[:user][:email] = email
+    user = User.find_by_email(email)
+    if not user.nil?
```

Comment:
makes sense thanks for the review, i would like someone with merging powers to let me know if they would be happy to merge this in before adding more code :)